portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,31 +23,91 @@
 import java.util.List;
 import com.google.gson.Gson;
 
+//datastore imports
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+//query imports
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> comments = new ArrayList<>();
+  private int maxComments = 3;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = new Gson().toJson(comments);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<String> comments = new ArrayList<>();
+    System.out.println(results.asIterable());
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("comment");
+      comments.add(comment);
+    }
+
+    // only show number of comments specified
+    String json;
+    if (comments.size() < maxComments){
+        json=new Gson().toJson(comments);
+    } else {
+        json=new Gson().toJson(comments.subList(0,maxComments));
```

Comment:
`json = new Gson()` for spacing consistency

Reply:
got it